npm/pkg/dataplane/ipsets/ipsetmanager.go

```diff
@@ -23,38 +24,49 @@ func (iMgr *IPSetManager) exists(name string) bool {
 	return ok
 }
 
-func NewIPSetManager() IPSetManager {
-	return IPSetManager{
-		setMap:      make(map[string]*IPSet),
-		dirtyCaches: make(map[string]struct{}),
+func NewIPSetManager() *IPSetManager {
+	iMgr := IPSetManager{
+		setMap: make(map[string]*IPSet),
 	}
+	iMgr.resetCaches()
+	return &iMgr
+}
+
+func (iMgr *IPSetManager) resetCaches() {
+	iMgr.dirtyCaches = make(map[string]struct{})
+	iMgr.deleteCache = make(map[string]string)
```

Comment:
We need the hashedName to make deletes, so then we would need to change the dirtyCache to be a map[string]string of set names to hashed names. This would use more memory than using a separate delete cache since we would need to specify a string for everything in the dirty cache